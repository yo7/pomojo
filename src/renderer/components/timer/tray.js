import {ipcRenderer} from 'electron'

export const initialize = () => update('')
export const update = text => ipcRenderer.send('update-timer', text)
