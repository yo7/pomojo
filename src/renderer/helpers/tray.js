import {ipcRenderer} from 'electron'

export const initializeTray = () => updateTray('')
export const updateTray = text => ipcRenderer.send('updateTray-timer', text)
