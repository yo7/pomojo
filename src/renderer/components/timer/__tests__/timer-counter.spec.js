import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import TimerCounter from '../timer-counter.vue'
import * as tray from '../tray'

describe('TimerCounter', () => {
  it('displays counter', () => {
    const wrapper = shallow(TimerCounter, {
      propsData: {seconds: 1499}
    })
    expect(wrapper.text()).toContain('24:59')
  })

  it('initalize tray on mounted', () => {
    jest.spyOn(tray, 'initialize')
    shallow(TimerCounter, {
      propsData: {seconds: 1500}
    })
    expect(tray.initialize).toHaveBeenCalled()
  })

  it('renders correctly', () => {
    const renderer = createRenderer()
    const wrapper = shallow(TimerCounter, {
      propsData: {seconds: 1500}
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error()
      }
      expect(str).toMatchSnapshot()
    })
  })
})
