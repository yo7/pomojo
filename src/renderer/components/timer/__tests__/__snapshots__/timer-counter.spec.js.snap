// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerCounter renders correctly 1`] = `
<div class="timer-counter">
    25:00
</div>
`;
